### PR TITLE
Bugfix/cmake

### PR DESCRIPTION
**Pull request by besser82**
As stated by the commit messages.

### CODE REVIEW

**Comment by vigsterkr** [comment]
thnx, travis is crying about:
```
CMake Error at test/CMakeLists.txt:35 (target_link_libraries):
  Target "catch-lib" of type UTILITY may not be linked into another target.
  One may link only to STATIC or SHARED libraries, or to executables with the
  ENABLE_EXPORTS property set.
```
see https://travis-ci.org/shogun-toolbox/tflogger/builds/380164907#L575

**Comment by besser82** [comment]
Should be fixed now.  I updated external Catch as well, so the undefined references should be fixed now, too.

If you're fine with merging, let me know and I'll merge on cli.

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -1,34 +1,40 @@
...
       tflogger_test.cc
       )
-target_include_directories(tfloggerTest PRIVATE SYSTEM ${CATCH_INCLUDE_DIRS})
+
+if(NOT TARGET Catch2::Catch)
+  message(STATUS "Catch will be downloaded when ${CMAKE_PROJECT_NAME} is built")
+  ExternalProject_Add(catch-lib
+        PREFIX ${THIRD_PARTY_PATH}/Catch
+        #--Download step--------------
+        URL https://github.com/catchorg/Catch2/archive/v1.12.2.zip
```
Comment (vigsterkr) [inline test/CMakeLists.txt]: shouldn't this be upgraded to actually 2.x... if we are checking for `Catch2` in the root cmake?

Comment (besser82) [reply test/CMakeLists.txt]: Okie, done that.  Added needed adjustments in that commit as well.

**Comment by besser82** [comment]
All stuff should be adressed now.  Any objections against releasing this as v0.1.1?

**Pull request merged**